crates/bevy_ecs/src/system/system.rs

```diff
@@ -37,6 +39,24 @@ pub trait System: Send + Sync + 'static {
     }
     fn apply_buffers(&mut self, world: &mut World);
     fn initialize(&mut self, _world: &mut World);
+    fn check_system_counter(&mut self, global_system_counter: u32);
 }
 
 pub type BoxedSystem<In = (), Out = ()> = Box<dyn System<In = In, Out = Out>>;
+
+pub(crate) fn check_system_counter_impl(
+    counter: &mut u32,
+    global_system_counter: u32,
+    system_name: &str,
+) {
+    let counter_age = global_system_counter.wrapping_sub(*counter);
+    let max_age = (u32::MAX / 4) * 3;
+    // Clamp to max age
+    if counter_age > max_age {
+        warn!(
+            "System '{}' was not run for too long, it may fail to detect changes.",
```

Comment:
This warning is unclear, especially for non-native speakers.

Instead: "Too many intervening systems have run since the last time System '{}' was last run; it may fail to detect changes.",